pkg/kube/wait.go

```diff
@@ -54,6 +72,9 @@ func (w *waiter) waitForResources(created ResourceList) error {
 		for _, v := range created {
 			ready, err := w.c.IsReady(ctx, v)
 			if !ready || err != nil {
+				if isServiceUnavailable(err) {
```

Comment:
There might be a situation where etcd doesn't recover forever or for  `x` amount of time (which may or may not be not tolerable for different customers using helm) and we end up waiting on this loop forever or `x` amount of time  because of this logic. What do you think ?
Shall we limit the number of retry attempts based on time factor and give up after that ?

Reply:
Never mind - I see that timeout is already set above - https://github.com/helm/helm/blob/ebc79fa807f29b984e090f0071b640f7347937cf/pkg/kube/wait.go#L68-L69

Reply:
yep.